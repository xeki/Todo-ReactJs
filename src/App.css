.App {
  text-align: center;
}
.container{
  width: 600px;
  margin: auto;
}
.counters {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  width: 450px;
  margin: 10px 0 20px 5px;
}
.counter-blocks {
  flex: 1;
  flex-basis: 200px;
  text-align: center;
}
.controls{
  display: flex;
  flex-direction: row;
  width: 450px;
  flex-wrap: nowrap;
  margin: 0 0 20px 5px;
}
.control-block{
  flex: 1;
  text-align: center;
}

li {
  width: 450px;
  display: flex;
  margin: 5px;
  flex-wrap: nowrap;
  background: gray;
  border-radius: 5px;
  min-height: 40px;
  vertical-align: middle;
  line-height: 40px;
}
.list-checkbox{
  flex: 1;
  text-align: center;
}
.list-taskname{
  flex: 2;
  text-align: center;
}
.list-button{
  flex: 1;
  text-align: center;
}

.create-button, .create-button:hover {
  background: navy;
  border: 2px solid navy;
  color: white;
  border-radius: 10px;
  outline: none;
}
.create-button:focus {
  outline: 1px solid #blue;
}
.delete-button {
  color: white;
  font-weight: 400;
  background: red;
  border-radius: 50%;
  border: 2px solid red;
}
input[type='text']{
  width: 200px;
  border-radius: 5px;
  border: 2px solid gray;
}
input[type='text']:focus, input[type='text']:active{
  outline: 1px solid blue;
}
* {
  font-family: comic sans ms;
  font-size: 16px;
}
