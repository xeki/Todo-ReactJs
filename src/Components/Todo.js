import React, { Component } from 'react';

class Todo extends Component {
constructor(props) {
  super(props);
  this.state = { uncheckedTask: 0, taskList: [] };
}
handleNewTask = () => {
  const task = document.querySelector('#task');
  let { uncheckedTask } = this.state;
  if (task.value) {
    const taskList = [...this.state.taskList, {task: task.value, checked: false }];
    uncheckedTask ++;
    task.value = '';
    this.setState({ taskList, uncheckedTask });
  }
}

deleteTask = (index) => {
  const checked = this.state.taskList[index].checked;
  const uncheckedTask = checked ? this.state.uncheckedTask : this.state.uncheckedTask -1;
  const taskList = [ ...this.state.taskList];
  taskList.splice(index,1);
  this.setState({ taskList, uncheckedTask });
}
handleCheckbox = (e) =>{
  let uncheckedTask = this.state.uncheckedTask;
  let { taskList } = this.state;
  taskList[e.target.dataset.index].checked = e.target.checked;
  if (e.target.checked) {
    uncheckedTask--;
  } else {
    uncheckedTask ++;
  }
  this.setState({ uncheckedTask, taskList });
}
renderTasks = () => {
  return this.state.taskList.map((task, index) => {
    return (
      <li key={index}>
        <div className='list-checkbox'>
            <input data-index={index} type='checkbox' checked={task.checked} onChange={this.handleCheckbox}/>
        </div>
        <div className='list-taskname'>
            <span> {task.task} </span>
       </div>
       <div className='list-button'>
            <button className='delete-button' name='delete' onClick={() => this.deleteTask(index)}> X </button>
      </div>
     </li>
    );
  })
}
  render() {
    return (
      <div className='container'>
          <div className='counters'>
              <div className='counter-blocks'> Tasks: {this.state.taskList.length} </div>
              <div className='counter-blocks'> Unchecked Tasks: {this.state.uncheckedTask} </div>
          </div>
          <div className='controls'>
            <div className='control-block'>
                <input type='text' name='task' placeholder='Task' id='task' />
            </div>
            <div className='control-block'>
                <button className='create-button' name='submit' onClick={this.handleNewTask}> Create Task </button>
            </div>
          </div>
          {this.renderTasks()}
    </div>
    );
  }

}

export default Todo;
