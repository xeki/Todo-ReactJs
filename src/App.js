import React from 'react';
import Todo  from './Components/Todo';
import './App.css';

function App() {
  return (
    <div>
      <Todo />
    </div>
  );
}

export default App;
