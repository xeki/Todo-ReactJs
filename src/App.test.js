import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Todo  from './Components/Todo';

describe('Functionality test suit', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches snapshot', () => {
  const todo = renderer.create(
    <Todo />
  );
  expect(todo.toJSON()).toMatchSnapshot();
});

it('should display initial state', () => {
  const todo = renderer.create(
    <Todo />
  );
  const instance = todo.getInstance();
  expect(instance.state.taskList.length).toBe(0);
});

});
