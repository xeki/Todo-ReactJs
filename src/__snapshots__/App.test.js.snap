// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Functionality test suit matches snapshot 1`] = `
<div>
  <div
    className="counters"
  >
    <div
      className="counter-blocks"
    >
       Tasks: 
      0
       
    </div>
    <div
      className="counter-blocks"
    >
       Unchecked Tasks: 
      0
       
    </div>
  </div>
  <div
    className="controls"
  >
    <div
      className="control-block"
    >
      <input
        id="task"
        name="task"
        placeholder="Task"
        type="text"
      />
    </div>
    <div
      className="control-block"
    >
      <button
        className="create-button"
        name="submit"
        onClick={[Function]}
      >
         Create Task 
      </button>
    </div>
  </div>
</div>
`;
